server/controllers/message.js

```diff
@@ -0,0 +1,164 @@
+    return res.status(500).json({ error: 'Could not create conversation' });
+  }
+});
+
+// @route POST /:id
+// Create a new message
+exports.newMessage = asyncHandler(async (req, res) => {
+  const { type, content } = req.body;
+  const recipient = req.params.id;
+  let decoded = decodeToken(req.cookies.token);
+  const userId = decoded.id;
+  try {
+    const message = await Message.create({
+      sender: userId,
+      recipient,
+      sendDate: Date.now(),
+      content,
+      type,
+    });
+    await Conversation.findOneAndUpdate(
+      { $and: [{ participants: { $in: userId } }, { participants: { $in: recipient } }] },
+      { $push: { messages: message }, mostRecentMsg: message, $inc: { unReadMsgs: 1 } },
+    );
+    res.status(201).json({
+      success: {
+        message: {
+          id: message._id,
+          content: message.content,
+          sendDate: message.sendDate,
+          read: message.read,
+          type: message.type,
+        },
+      },
+    });
+  } catch (error) {
+    return res.status(500).json({ error: 'Could not send mesage' });
+  }
+});
+
+// @route GET /conversation/all
+// Populate chat links with all conversations
+exports.getConversations = asyncHandler(async (req, res) => {
+  let decoded = decodeToken(req.cookies.token);
+  const userId = decoded.id;
+
+  try {
+    const conversations = await Conversation.find({
+      $and: [{ participants: { $in: userId } }, { deleted: false }],
+    }).populate('mostRecentMsg');
+    res.status(200).json({
+      success: {
+        conversations,
+      },
+    });
+  } catch (error) {
+    return res.status(500).json({ error: 'Could not get conversation' });
+  }
+});
+
+// @route DELETE /conversation/delete/:id
+// Delete conversation
+exports.deleteConversation = asyncHandler(async (req, res) => {
+  let conversationId = req.params.id;
+  try {
+    await Conversation.findOneAndUpdate({ _id: conversationId }, { deleted: true });
+    res.status(200).json({ success: true });
+  } catch (error) {
+    return res.status(500).json({ error: 'Could not delete conversation' });
+  }
+});
+
+// @route PATCH /messages/read/:id
+// Mark all messages in a converstion as read
+exports.readMessages = asyncHandler(async (req, res) => {
+  const conversationId = req.params.id;
+  try {
+    const conversation = await Conversation.findOne({ _id: conversationId }).populate('messages');
+    conversation.messages.forEach(async (message) => {
+      if (message.read === false) {
+        message.read = true;
+        await message.save();
+      }
+    });
+    await Conversation.findOneAndUpdate({ _id: conversationId }, { unreadMsgs: 0 });
+    res.status(200).json({ success: true });
+  } catch (error) {
+    return res.status(500).json({ error: 'Could not mark messages in conversation as read' });
+  }
+});
+
+// @route GET /conversation/show/:id
+// Get a conversation with messages with a another user
+exports.getConversation = asyncHandler(async (req, res) => {
```

Comment:
Maybe you can add the above logic to this controller so that we can set messages to read when we get the messages within a conversation?

Reply:
OK